mergify_engine/tasks/engine/__init__.py

```diff
@@ -161,24 +161,73 @@ def run(event_type, data):
         return
 
     with github.get_client(owner, repo, installation) as client:
-        _run(client, event_type, data)
+        pull = get_github_pull_from_event(client, event_type, data)
 
+        if not pull:  # pragma: no cover
+            LOG.info(
+                "No pull request found in the event %s, ignoring",
+                event_type,
+                gh_owner=client.owner,
+                gh_repo=client.repo,
+            )
+            return
+
+        # Get a fresh pull request view
+        # Everything after this must handle batching per pull request
+        sources = [{"event_type": event_type, "data": data}]
+        ctxt = context.Context(client, pull)
+
+        issue_comment_sources = []
+
+        for source in sources:
+            if source["event_type"] == "issue_comment":
+                issue_comment_sources.append(source)
+            elif not is_source_obsolete(ctxt.pull, source):
+                ctxt.sources.append(source)
+
+        commands_runner.spawn_pending_commands_tasks(ctxt)
+
+        if issue_comment_sources:
+            for source in issue_comment_sources:
```

Comment:
Are we sure that the sources are ordered here?

Reply:
Because of gevent, nothing is ordered already. @sileht

Reply:
Alright, it might not be a big deal for now but they should be at some point.